### ISSUE TITLE
Report connection status to QualityOracle

### ISSUE THREAD

**Issue by zolia** [mysteriumnetwork/node#1298]
after #1297 is implemented report connection status to QualityOracle. We could do reporting from both sides (from consumer and from provider). QualityOracle should update Provider quality metrics by matching reported status from both sides.

Currently consumer reports on failed connects, but not all failed cases are accounted for. After Consumer gets connected event he still needs to be sure that actual traffic gets transferred. 

Sent Connection status from Provider could contain and determined NAT traversal state for given connection moment. These can be two separate messages tho.

**Comment by soffokl** [comment]
Is this a duplicate of https://github.com/mysteriumnetwork/node/issues/891?

**Comment by anjmao** [comment]
Yes they looks similar, let's continue here as this task is a follow up of #1297.

**Comment by anjmao** [comment]
Currently we have these metrics for session in prometheus.

```
session_event{event="Connecting"}
session_event{event="Connected"}
session_event{event="Created"}
session_event{event="Disconnecting"}
session_event{event="Ended"}
session_event{event="NotConnected"}
```

In node session_event metric is aggregated from two different events (EventTopicState = "State" and EventTopicSession = "Session") which makes it confusing to understand the difference in Connected or Created and Disconnecting and Ended statuses. To be more clear in node code only Created and Ended are actual session events and all others are connection state events.

[Here](https://github.com/mysteriumnetwork/node/blob/master/core/connection/status.go) we have connection status state:

```go
// State represents list of possible connection states
type State string

const (
	// NotConnected means no connection exists
	NotConnected = State("NotConnected")
	// Connecting means that connection is startCalled but not yet fully established
	Connecting = State("Connecting")
	// Connected means that fully established connection exists
	Connected = State("Connected")
	// Disconnecting means that connection close is in progress
	Disconnecting = State("Disconnecting")
	// Reconnecting means that connection is lost but underlying service is trying to reestablish it
	Reconnecting = State("Reconnecting")
	// Unknown means that we could not map the underlying transport state to our state
	Unknown = State("Unknown")
	// Canceled means that connection initialization was started, but failed never reaching Connected state
	Canceled = State("Canceled")
)
```

And separate connection session status [here](https://github.com/mysteriumnetwork/node/blob/master/core/connection/event.go#L38)
```go
const (
	// SessionCreatedStatus represents a session creation event
	SessionCreatedStatus = "Created"
	// SessionEndedStatus represents a session end
	SessionEndedStatus = "Ended"
)
```
In the end they are merged into session_event metric.

Few possible ways I see how to solve it:

1. Move SessionCreatedStatus and SessionEndedStatus into State so it become:
```go
SessionCreated = State("SessionCreated")
SessionEnded = State("SessionEnded")
```
2. Rename metric to connection_event which can then have both connection and session states.

In the end metric will like this:

```
connection_event{event="Connecting"}
connection_event{event="Connected"}
connection_event{event="SessionCreated"}
connection_event{event="Disconnecting"}
connection_event{event="SessionEnded"}
connection_event{event="NotConnected"}
connection_event{event="IPNotChanged"} // My new metric.
connection_event{event="ConnectionFailed"} // My new metric.
```
3. Remove event topic **Session** as State topic already contains SessionInfo.

**Comment by anjmao** [comment]
As we clarified with @Waldz session_event Created means that session is created and Connected means that underlying session vpn tunnel is connected. I will keep everything as is and only add new metrics to connection State.

**Comment by eugenegoncharuk** [comment]
@soffokl How do You think, is this still relevant?

**Comment by soffokl** [comment]
yes, let's keep it open.
